libfwupd/fwupd-enums.h

```diff
@@ -94,6 +94,8 @@ typedef enum {
  * @FWUPD_DEVICE_FLAG_WILL_DISAPPEAR:		Device will disappear after update and can't be verified
  * @FWUPD_DEVICE_FLAG_CAN_VERIFY:		Device checksums can be compared against metadata
  * @FWUPD_DEVICE_FLAG_CAN_VERIFY_IMAGE:		Image can be dumped from device for verification
+ * @FWUPD_DEVICE_FLAG_DUAL_IMAGE:		Device update architecture uses A/B partitions for updates
+ * @FWUPD_DEVICE_FLAG_RESILIENT_RECOVERY:	In flashing mode device will only accept intended payloads
```

Comment:
I like `DUAL_IMAGE`, but not convinced by `RESILIENT_RECOVERY` -- isn't "flashing mode" more generally known as "bootloader mode"? Maybe something like `BOOTLOADER_RECOVERY`?

Reply:
Well not all devices need to go into bootloader mode to flash.  None of the stuff in dell-dock does for example.

Reply:
Something else missing from these flags that I think is useful to a client in making decisions:
"Device is usable while flashing".  This might be a 3rd flag and then you can have this one become BOOTLOADER_RECOVERY

Reply:
> Device is usable while flashing

I think this is a really important flag. I can certainly use this in gnome-software too.

Reply:
The difficult thing to message with a flag like that is:
* whether the device is usable the entire time (controls it's own reboot)
* has a quick restart cycle but is usable most of the time (Thunderbolt falls in this category)
* requires a user to reboot the device manually somehow

Reply:
> whether the device is usable the entire time

that's the one i was thinking about; the ones that reboot we should still tell the user to stop using them.

Reply:
OK added it and reworded some more again, take another look.